src/Uart/Uart.c

```diff
@@ -189,22 +174,25 @@ void
 Uart_shutdown(Uart* const uart)
 {
     rtems_interrupt_vector_disable(interruptNumber(uart->id));
-    rtems_interrupt_entry_remove(interruptNumber(uart->id),
-                                 &(uart->interruptData.rtemsInterruptEntry));
+    rtems_interrupt_entry_remove(interruptNumber(uart->id), &(uart->interruptData.rtemsInterruptEntry));
     uart->reg->control = 0;
     uart->reg->status = 0;
     uart->interruptData.sentBytes = 0;
-    uart->errorFlags = (Uart_ErrorFlags){ 0 };
+    uart->errorFlags = (Uart_ErrorFlags){0};
 }
 
 void
 Uart_init(Uart_Id id, Uart* const uart)
 {
     uart->id = id;
+#ifndef UNIT_TESTS
     uart->reg = getAddressBase(id);
-    uart->interruptData.rtemsInterruptEntry = (rtems_interrupt_entry){ 0 };
+#else
+    uart->reg = malloc(sizeof(UartRegisters_t));
```

Comment:
I don't like this malloc, I would prefer if code was not aware of being tested. But for now, I am more concerned about the schedule, so let it be. BUT, I think it would be less scary if we used ifdef instead of ifndef, then the malloc would be just live after UNIT_TEST. Also, consider renaming UNIT_TEST to MOCK_REGISTERS or someting more precise.